src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+        user.setAge(20);
+    }
+
+    @AfterEach
+    void afterEach() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        user.setLogin(" ");
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for login [ ]");
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        user.setPassword(" ");
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for password [ ]");
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for age null");
+    }
+
+    @Test
+    void register_ageLessThanEighteen_notOk() {
+        user.setAge(14);
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for age [14]");
+    }
+
+    @Test
+    void register_tooBigAge_notOk() {
+        user.setAge(150);
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for age [150]");
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for password null");
+    }
+
+    @Test
+    void register_passwordShorterThanSixChars_notOk() {
+        user.setPassword("bob1");
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for password [bob1]");
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for login null");
+    }
+
+    @Test
+    void register_userIsNull_notOk() {
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(null);
+        }, "InvalidRegistrationDataException expected for user null");
+    }
+
+    @Test
+    void register_idIsNotNull_notOk() {
+        user.setId(50L);
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for id which is not null");
+    }
+
+    @Test
+    void register_addTwoSameUsers_notOk() {
+        registrationService.register(user);
```

Comment:
You can’t use method `register()` here because you testing this method. You need to add a user to the base like this: `Storage.people.add()`